packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.ts

```diff
@@ -50,21 +53,43 @@ export function create(context: RuleContext<MessageID, []>): RuleListener {
           continue;
         }
         const propsType = getConstrainedTypeAtLocation(services, props);
-        if (isTypeReadonlyLoose(services, propsType)) {
-          continue;
-        }
+        if (isTypeReadonly(services.program, propsType)) continue;
+        // Handle edge case where isTypeReadonly cant detect some readonly or immutable types
+        if (isTypeReadonlyLoose(services, propsType)) continue;
+        // @see https://github.com/Rel1cx/eslint-react/issues/1326
+        if (propsType.isClassOrInterface() && isClassOrInterfaceReadonlyLoose(checker, propsType)) continue;
         context.report({ messageId: "preferReadOnlyProps", node: props });
       }
     },
   };
 }
 
 function isTypeReadonlyLoose(services: ParserServicesWithTypeInformation, type: ts.Type): boolean {
-  if (isTypeReadonly(services.program, type)) return true;
   try {
     const im = getTypeImmutability(services.program, type);
     return isUnknown(im) || isImmutable(im) || isReadonlyShallow(im) || isReadonlyDeep(im);
   } catch {
     return true;
   }
 }
+
+// TODO: A comprehensive test is required to verify that it works as expected
+// @see https://github.com/Rel1cx/eslint-react/issues/1326
+function isClassOrInterfaceReadonlyLoose(checker: ts.TypeChecker, type: ts.Type) {
+  const baseTypes = type.getBaseTypes() ?? [];
+  const properties = type.getProperties();
+  if (properties.length === 0) {
+    return true;
+  }
+  if (baseTypes.length === 0) {
+    return properties.every((property) => isPropertyReadonlyInType(type, property.getEscapedName(), checker));
+  }
+  for (const property of properties) {
+    const propertyName = property.getEscapedName();
+    if (isPropertyReadonlyInType(type, propertyName, checker)) continue;
+    else if (baseTypes.length > 0) {
+      return baseTypes.every((heritageType) => isPropertyReadonlyInType(heritageType, propertyName, checker));
```

Comment:
Fixed in https://github.com/Rel1cx/eslint-react/commit/ee3fc704b4b935875e65cda2b3a858389191fd58